### DISCUSSION TITLE
Unable to patch the request.state

### DISCUSSION THREAD

**Discussion by ikarus-prashanth** [fastapi/fastapi#9668] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from unittest.mock import patch, MagicMock
from fastapi.testclient import TestClient
from main import app 

test_client = TestClient(app)

@patch('mymodule.get_session_information')
def test(mock_get_session_information):
    mock_session_info = SessionInformationResult(
        session_handle=SessionInformationResult.SessionHandle({"username_bool": True}),
        user_id="test_user_id",
        session_data={},
        expiry=81,
        access_token_payload={"username_bool": True},
        time_created=637,
    )
    mock_get_session_information.return_value = mock_session_info
    response = test_client.get("/isunthere")
    assert response.status_code == 200
    assert response.json() == True
```

### Description

I'm trying to test a FastAPI endpoint but I'm getting the fllowing eroor
`AttributeError: 'State' object has no attribute 'session'`
Whern the test endpoint is triggered, it is creating an object of the associated function and the original method is being invoked, not the mocked one!

### Operating System

Windows

### FastAPI Version

0.95.2

### Python Version

3.11.3

**Comment by ikarus-prashanth** [comment]
@odiseo0 @gonzacastro @JarroVGIT Could you please look into this!

**Comment by YuriiMotov** [comment, marked as answer]
I guess `get_session_information` is imported to one of modules (maybe `main`) and is used there. So, you need to patch this variable in that module, not in the module where it's originally defined:
(Assuming it's being used in `main`)
```python
@patch('main.get_session_information')
```

**Comment by YuriiMotov** [comment]
I guess `get_session_information` is imported to one of modules (maybe `main`) and is used there. So, you need to patch this variable in that module, not in the module where it's originally defined:
(Assuming it's being used in `main`)
```python
@patch('main.get_session_information')
```
